pkg/agent/agent.go

```diff
@@ -19,48 +21,51 @@ type Config struct {
 
 // Agent maintains the state required for executing an agent command
 type Agent struct {
-	env    runtime.Environment
-	config *Config
+	env           runtime.Environment
+	sc            <-chan os.Signal
+	profileCloser io.Closer
 }
 
-// BuildAgent builds a instance of an agent
-func BuildAgent(env runtime.Environment, config *Config) *Agent {
-	return &Agent{
-		env:    env,
-		config: config,
+// BuildAndStart creates and starts a new instance of an agent.
+// Returned agent is guaranteed to be unique in the environment it is running, and will handle signals sent to the
+// process.
+// Callers must Stop the returned agent at the end of its lifecycle.
+func BuildAndStart(env runtime.Environment, config *Config) (*Agent, error) {
```

Comment:
I agree with the convenience but I'm not sure about the naming, mostly because of the `and` part. I would prefer calling it simply `Start`.  But this is not a major issue.